walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -114,4 +145,56 @@
 
     return possibleSlots;
   }
+
+  /**
+   * Find the intersection of two collections of slots. The intersection between two individual
+   * slots is the slot that is contained within the other one. If none of the two slots is contained
+   * within the other one, then their intersection is empty.
+   *
+   * @param mandatory A collection of {@code TimeRange}s that represents meeting slots for mandatory
+   *     attendees
+   * @param optional A collection of {@code TimeRange}s that represents meeting slots for optional
+   *     attendees
+   * @return The intersection of these collections as specified above.
+   */
+  public Collection<TimeRange> intersection(
+      Collection<TimeRange> mandatory, Collection<TimeRange> optional) {
+    Iterator<TimeRange> itMandatory = mandatory.iterator();
+    Iterator<TimeRange> itOptional = optional.iterator();
+
+    TimeRange currentMandatory = itMandatory.next();
+    TimeRange currentOptional = itOptional.next();
```

Comment:
This will throw an exception if `optional` is empty. I think this can occur if there are optional attendees in the request, but no optional slots are generated.

Either way, I would recommend adding an if to handle the edge case where one of them is empty.

Reply:
I check whether optional slots are empty here: 
![image](https://user-images.githubusercontent.com/41055600/89540708-16137a00-d806-11ea-8b2c-3eed6797bf6c.png)

The two conditions can't be true at the same time because if there are no optional attendees, there will be one slot returned, i.e. WHOLE_DAY. So the code always catches the case when there are no optional slots.